13-collabs.Rmd

```diff
@@ -5,6 +5,248 @@
+
+```{r, fig.alt="Ruby has filed a very large pull request with a very short PR description:  I updated the project and added new files. She then requested Avi to review this PR. This leaves Avi with very little information to go on and he is very confused. Avi comments: Ummm… I’m a bit confused. Can you explain the context of these changes? This is a lot for me to try to follow.  I’m also unsure what kind of feedback you are looking for. ", out.width = "100%", echo = FALSE}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g33bf0789107_300_2369")
+```
+
+Even if you end up being the only person who will review your own code, writing these things out is still very helpful and highly recommended. It can help you spot problems you might not have otherwise seen and generally help you document better for future you!
+
+## Characteristics of great pull requests and issues
+
+### There's plenty of context!
+
+What's the story behind around the changes you are proposing? Sometimes when we are in the thick of a project we can make the mistake of assuming everyone knows what we know. This can unfortunately leave a huge burden on your reviewer to try to follow a paper trail to try to understand what you are doing.
+
+Before sending off a review request, re-read your PR description and think about the perspective of your reviewer. Err on the side that they have no idea what is happening on the project (because sometimes this is the case!)
+
+Tell a short story to explain what lead to you making these changes including attempting to answer these questions:  
+
+- What is the problem that these changes will solve?
+- Do you have any URLs relevant issues or files you can share?
+- What inspired you to take this approach -- are there other things you tried?
+- Are there other pull requests related to this change?
+
+### Includes an explicit request for what kind of feedback is needed
+
+What would you like your reviewer to do with this pull request? Stating this explicitly can save both of you time in this code review.
+
+- Are you still in the early stages and looking for a bigger picture review? Let them know that before they waste their time digging into the code line-by-line.
+- Are you in the later stages and looking for detailed nit-picky review?
+- Are you looking for feedback on the results or methods?
+
+### Points out questionable areas that need extra attention
+
+Are there specific areas of the code you are having trouble with or are unsure about? Send a link to the [specific lines in GitHub](https://stackoverflow.com/questions/23821235/how-to-link-to-specific-line-number-on-github) you are asking about. Are there results that are surprising, confusing, or [smell wrong](https://github.com/jennybc/code-smells-and-feels#code-smells-and-feels)?  
+
+Be sure to detail what you have dug into and tried at this point for any problematic points.   
```

Comment:
```suggestion
Be sure to detail what you have dug into and tried at this point for any problematic points.  This can help the reviewer to avoid suggesting potential solutions that the author has already tried. 
```